assembly/frame/executive.ts

```diff
@@ -0,0 +1,185 @@
+	 * except state-root.
+     */
+    export function finalizeBlock(): HeaderType {
+        System.noteFinishedExtrinsics();
+        System.computeExtrinsicsRoot();
+        return System.finalize() as HeaderType;
+    }
+    /**
+     * @description creates inherents from internal modules
+     * @param data inherents
+     */
+    export function createExtrinsics(data: IInherentData): u8[] {
+        const timestamp: InherentType = Timestamp.createInherent(data);
+        const aura = Aura.createInherent(data);
+        return System.ALL_MODULES.concat(timestamp.toU8a()).concat(aura);
+    }
+
+    /**
+     * @description Apply Extrinsics
+     * @param ext extrinsic
+     */
+    export function applyExtrinsic(ext: u8[]): u8[] {
+        const encodedLen = BytesReader.decodeInto<CompactInt>(ext);
+        const result = Executive.applyExtrinsicWithLen(ext, encodedLen.value as u32);
+        // if applying extrinsic succeeded, notify System about it
+        if(Utils.areArraysEqual(result, ResponseCodes.SUCCESS)){
+            System.noteAppliedExtrinsic(ext);
+        }
+        return result;
+    }
+    
+    /**
+     * @description Orchestrating function for applying extrinsic
+     * @param ext extrinsic
+     * @param encodedLen length
+     * @param encoded encoded extrinsic
+     */
+    export function applyExtrinsicWithLen(ext: u8[], encodedLen: u32): u8[]{
+        switch(encodedLen){
+            case ExtrinsicType.Inherent:{
+                const inherent: IInherent = BytesReader.decodeInto<InherentType>(ext);
+                return Timestamp.applyInherent(<InherentType>inherent)
+            }
+            case ExtrinsicType.SignedTransaction:{
+                const signedTransaction: IExtrinsic = BytesReader.decodeInto<SignedTransactionType>(ext);
+                return Balances.applyExtrinsic(<SignedTransactionType>signedTransaction);
+            }
+            default:{
+                return ResponseCodes.CALL_ERROR;
+            }
+        }
+    }
+
+    /**
+     * @description Execute given extrinsics and take care of post-extrinsics book-keeping
+     * @param extrinsics byte array of extrinsics 
+     */
+    export function executeExtrinsicsWithBookKeeping(extrinsics: IExtrinsic[]): void{
+        for(let i=0; i<extrinsics.length; i++){
+            Executive.applyExtrinsic(extrinsics[i].toU8a());
+        }
+        System.noteFinishedExtrinsics();
+    }
+
+    /**
+     * @description Initial transaction validation
+     * @param source source of the transaction (external, inblock, etc.)
+     * @param utx transaction
+     */
+    export function validateTransaction(utx: ISignedTransaction): u8[] {
+        const from: AccountIdType = BytesReader.decodeInto<AccountIdType>(utx.getFrom().toU8a());
+        const transfer = utx.getTransferBytes();
+        Log.info("tx bytes: " + transfer.toString());
+
+        if(!Crypto.verifySignature(<SignatureType>utx.getSignature(), transfer, from)){
```

Comment:
Yes, because the return type of `utx.getSignature()` is `Codec`, unless I explicitly cast it to `SignatureType` it will assume it as `Codec` type, and IDE will warn `Argument of type 'Codec' is not assignable to parameter of type 'ISignature'.`;